testplan/common/entity/base.py

```diff
@@ -1125,6 +1125,8 @@ def _execute_step(self, step, *args, **kwargs):
         res = None
         try:
             res = step(*args, **kwargs)
+        except RuntimeError as exc:
```

Comment:
This overall looks very good, if you just want to propagate the exception when there's a problem with runpath conflict, maybe define a specialized exception class for that, RuntimeError is a bit generic.